tx.go

```diff
@@ -114,26 +122,44 @@ func AttachSenderToTransaction(sender *skademlia.Keypair, tx Transaction, parent
 	tx.SenderSignature = edwards25519.Sign(sender.PrivateKey(), tx.Marshal())
 
 	tx.rehash()
-
-	return tx
 }
 
-func (t *Transaction) rehash() *Transaction {
+func (t *Transaction) rehash() {
 	t.ID = blake2b.Sum256(t.Marshal())
 
-	buf := make([]byte, 0, SizeAccountID+len(t.ParentIDs)*SizeTransactionID)
-	buf = append(buf, t.Sender[:]...)
-	for _, parentID := range t.ParentIDs {
-		buf = append(buf, parentID[:]...)
-	}
+	// Calculate the new seed.
+	{
+		hasher, err := blake2b.New(32, nil)
+		if err != nil {
+			panic(err) // should never happen
+		}
 
-	t.Seed = blake2b.Sum256(buf)
-	t.SeedLen = byte(prefixLen(t.Seed[:]))
+		_, err = hasher.Write(t.Sender[:])
+		if err != nil {
+			panic(err) // should never happen
```

Comment:
If it should never happen, then write instead _, _ = hasher.Write(t.Sender[:]).

Reply:
Same goes for other error-check statements within `rehash()`.

Reply:
i would put `log` even in case it will "never" happen.

Reply:
If something that should never happen does happen, it should be a fatal error. Panic in this case.

Reply:
I'd argue just explicitly ignoring via _, _ = hasher.Write(t.Sender[:]). The documentation states that an error will never be thrown. A panic nor an error in this case should never have to occur. If needed, a comment can be appended to emphasize this.